src/addressResolvers/farcaster.ts

```diff
@@ -0,0 +1,107 @@
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { graphQlCall, Address, Handle, FetchError, isSilencedError, isEvmAddress } from './utils';
+
+export const NAME = 'Farcaster';
+const FNAMES_API_URL = 'https://fnames.farcaster.xyz/transfers?name=';
+const NEYNAR_API_URL = 'https://api.neynar.com/v2/farcaster/user/';
+const API_KEY = process.env.NEYNAR_API_KEY ?? '';
+
+
+interface UserDetails {
+  username: string;
+  verified_addresses: {
+    eth_addresses: string[];
+    sol_addresses: string[];
+  };
+  pfp_url: string;
+}
+
+interface ApiResponse {
+  [address: string]: UserDetails[];
+}
+
+interface UserResult {
+  username?: string;
+  eth_addresses?: string[];
+  sol_addresses?: string[];
+  pfp_url?: string;
+}
+
+
+async function fetchData<T>(url: string, method: string = 'GET'): Promise<T> {
+  const headers = {
+    Accept: 'application/json',
+    api_key: API_KEY
+  };
+  const response = await fetch(url, { method, headers });
+  if (!response.ok) {
+    throw new Error(`Failed to fetch data from the API. Status: ${response.status}`);
+  }
+  return response.json() as Promise<T>;
+}
+
+export async function lookupAddresses(
+  addresses: string[]
+): Promise<{ [key: string]: UserResult | string }> {
+  const results: { [key: string]: UserResult | string } = {};
+  try {
+    const addressesQuery = addresses.join(',');
```

Comment:
Addresses should be filtered, like in 

https://github.com/snapshot-labs/stamp/blob/39d9f802086df8830183130ce84dbf8020860c40/src/addressResolvers/lens.ts#L42

https://github.com/snapshot-labs/stamp/blob/39d9f802086df8830183130ce84dbf8020860c40/src/addressResolvers/lens.ts#L31-L33

So that we filter out non supported addresses format (such as starknet 64 chars addresses).

Reply:
Why skipping on this filtering ?

Reply:
isEVMAddress is always expecting an address array, I will make a small change in my code to send it as an array and  eliminate the viem lib